### ISSUE TITLE
When TOC is empty, "Table of contents" still exists.

### ISSUE THREAD

**Issue by nzhangac** [squidfunk/mkdocs-material#2586]
Hi, when I build with mkdocs-material, I find that when TOC is empty, "Table of contents" still exists.

![My-Docs](https://user-images.githubusercontent.com/48249362/115445950-17c48700-a249-11eb-9c45-dfa0f09191f5.png)

And I find that change 

https://github.com/squidfunk/mkdocs-material/blob/b3a39184195d246e0aa2b7b16cce624f2f584a0b/src/partials/toc.html#L41

to

```python
{% if toc %} 
```

will fix it.

**Comment by squidfunk** [comment]
Does this happen with 7.1.1? Might be a regression.

**Comment by nzhangac** [comment]
> Does this happen with 7.1.1? Might be a regression.

No, 7.1.1 works well. 

"partials/toc.html" comparison between 7.1.1 and 7.1.2. Left side is from 7.1.1.

```
{#-                                                             {#-
  This file was automatically generated - do not edit             This file was automatically generated - do not edit
-#}                                                             -#}
{% import "partials/language.html" as lang with context %}      {% import "partials/language.html" as lang with context %}
<nav class="md-nav md-nav--secondary" aria-label="{{ lang.t('   <nav class="md-nav md-nav--secondary" aria-label="{{ lang.t('
  {% set toc = page.toc %}                                        {% set toc = page.toc %}
  {% if toc | first is defined and "\x3ch1 id=" in page.conte |   {% set first = toc | first %}
    {% set toc = (toc | first).children %}                    |   {% if first is defined and first.level == 1 %}
                                                              >     {% set toc = first.children %}
  {% endif %}                                                     {% endif %}
  {% if toc | first is defined %}                             |   {% if toc is iterable %}
    <label class="md-nav__title" for="__toc">                       <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>                      <span class="md-nav__icon md-icon"></span>
      {{ lang.t("toc.title") }}                                       {{ lang.t("toc.title") }}
    </label>                                                        </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-       <ul class="md-nav__list" data-md-component="toc" data-md-
      {% for toc_item in toc %}                                       {% for toc_item in toc %}
        {% include "partials/toc-item.html" %}                          {% include "partials/toc-item.html" %}
      {% endfor %}                                                    {% endfor %}
    </ul>                                                           </ul>
  {% endif %}                                                     {% endif %}
</nav>                                                          </nav>
```

**Comment by squidfunk** [comment]
Thanks for testing. Yes we changed the way the table of contents is rendered to fix #2563, so it's definitely a regression. I'll look into it asap.

**Comment by squidfunk** [comment]
Fixed in 319db2d57 – thanks for time you took to investigate!

**Comment by squidfunk** [comment]
Released as part of 7.1.3.
